.github/workflows/group-35-to-311-ci.yml

```diff
@@ -0,0 +1,84 @@
+name: Moodle plugin CI group for Moodle 3.5-3.11
+
+on:
+  workflow_call:
+    inputs:
+      extra_plugin_runners:
+        type: string
+      disable_behat:
+        type: boolean
+      disable_phplint:
+        type: boolean
+      disable_phpunit:
+        type: boolean
+      disable_grunt:
+        type: boolean
+
+jobs:
+  setup:
+    name: 3.5 - 3.11
+    env:
+      IGNORE_PATHS: tests/fixtures
+    runs-on: 'ubuntu-latest'
+
+    services:
+      postgres:
+        image: postgres:9.6
+        env:
+          POSTGRES_USER: 'postgres'
+          POSTGRES_HOST_AUTH_METHOD: 'trust'
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 3
+        ports:
+          - 5432:5432
+
+      mariadb:
+        image: mariadb:10.5
+        env:
+          MYSQL_USER: 'root'
+          MYSQL_ALLOW_EMPTY_PASSWORD: "true"
+        ports:
+          - 3306:3306
+        options: >-
+          --health-cmd="mysqladmin ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 3
+
+    strategy:
+      fail-fast: false
+      matrix:
+        database: ['mariadb', 'pgsql']
+        moodle-branch: ['MOODLE_35_STABLE', 'MOODLE_36_STABLE', 'MOODLE_37_STABLE', 'MOODLE_38_STABLE']
+        node: ['14.15']
+        php: ['7.1', '7.2']
+        include:
+          - {moodle-branch: 'MOODLE_36_STABLE', php: '7.3', node: '14.15', database: 'mariadb'}
+          - {moodle-branch: 'MOODLE_36_STABLE', php: '7.3', node: '14.15', database: 'pgsql'}
+          - {moodle-branch: 'MOODLE_37_STABLE', php: '7.3', node: '14.15', database: 'mariadb'}
+          - {moodle-branch: 'MOODLE_37_STABLE', php: '7.3', node: '14.15', database: 'pgsql'}
+          - {moodle-branch: 'MOODLE_38_STABLE', php: '7.3', node: '14.15', database: 'mariadb'}
+          - {moodle-branch: 'MOODLE_38_STABLE', php: '7.3', node: '14.15', database: 'pgsql'}
+          - {moodle-branch: 'MOODLE_39_STABLE', php: '7.2', node: '14.15', database: 'mariadb'}
+          - {moodle-branch: 'MOODLE_39_STABLE', php: '7.2', node: '14.15', database: 'pgsql'}
+          - {moodle-branch: 'MOODLE_39_STABLE', php: '7.3', node: '14.15', database: 'mariadb'}
+          - {moodle-branch: 'MOODLE_39_STABLE', php: '7.3', node: '14.15', database: 'pgsql'}
+          - {moodle-branch: 'MOODLE_310_STABLE', php: '7.2', node: '14.15', database: 'mariadb'}
+          - {moodle-branch: 'MOODLE_310_STABLE', php: '7.2', node: '14.15', database: 'pgsql'}
+          - {moodle-branch: 'MOODLE_310_STABLE', php: '7.3', node: '14.15', database: 'mariadb'}
+          - {moodle-branch: 'MOODLE_310_STABLE', php: '7.3', node: '14.15', database: 'pgsql'}
+          - {moodle-branch: 'MOODLE_311_STABLE', php: '7.3', node: '14.15', database: 'mariadb'}
+          - {moodle-branch: 'MOODLE_311_STABLE', php: '7.3', node: '14.15', database: 'pgsql'}
```

Comment:
can we trim the matrix a little? eg keep php 7.2 on maria on 3.9 and php 7.2 on postgres on 3.10 but remove a few of the other 3.10's

I'd like to have a rule of thumb in place for this and add it to the readme, something like across the matrix we need to do all php versions, and all db's but not all combinations in the older stables. Only do a full matrix on the LTS stables.

Reply:
Moved this into a new issue https://github.com/catalyst/catalyst-moodle-workflows/issues/9